### DISCUSSION TITLE
Brew is very slow updating/upgrading

### DISCUSSION THREAD

**Discussion by jorgeceballosgarcia** [Homebrew/discussions#4443] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.14
ORIGIN: https://github.com/Homebrew/brew
HEAD: ac0663ae5d2d4c76bb62363592d77dbad49ac30c
Last commit: 4 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: e57c9fc4ba4358a1d4748db4f685d114ea0f0ae7
Core tap last commit: 6 weeks ago
Core tap branch: master
Core tap JSON: 20 Apr 07:47 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.2KXLP2KtaF/org.xquartz:0
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.87.0 => /usr/bin/curl
macOS: 13.3.1-arm64
CLT: 14.3.0.0.1.1679647830
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  ansible@2.9
  go@1.18

Warning: Broken symlinks were found. Remove them with `brew cleanup`:
  /opt/homebrew/opt/kubectl@1.26
```

### Description of issue

Hello,

When I do any installation through brew or any update it can take more than an hour to update 2 or 3 packages.

I understand that this is not normal, do I need some type of configuration?

Regards

**Comment by SMillerDev** [comment]
What is it showing during that update? Because most take seconds for me.

**Reply by jorgeceballosgarcia** [reply]
Nothing.

For example this update starts at 12:40 and finished at 12:45 for only 1 package

![Screenshot 2023-04-20 at 12 41 38](https://user-images.githubusercontent.com/102051248/233342979-af3b850c-0cf3-42db-935c-f4415030baee.png)

![Screenshot 2023-04-20 at 12 45 05](https://user-images.githubusercontent.com/102051248/233343608-f6f216aa-59bc-4da0-b68c-f9a148b1241d.png)

Exists any log output from brew executions?

Regards

**Reply by SMillerDev** [reply]
So it's not the download that is slow right? It's pouring the bottle.

**Reply by jorgeceballosgarcia** [reply]
No, the download is very fast, and yes, the large time is for pouring the bottle.

**Reply by jorgeceballosgarcia** [reply]
From today:

> jocebal@jocebal-mac ~ % date && brew update && brew upgrade && date
> Fri Apr 21 12:09:09 CEST 2023
> Updated 3 taps (homebrew/services, homebrew/core and homebrew/cask).
> ==> New Formulae
> ada-url                                     amber                                       ata                                         chatblade                                   ffmpeg@5
> ==> New Casks
> active-trader-pro                                                                                              xtool-creative-space
> Fri Apr 21 12:10:40 CEST 2023

**Reply by jorgeceballosgarcia** [reply]
Hi @SMillerDev 

More logs from today:

> jocebal@jocebal-mac ~ % date; brew update; brew upgrade; date
> **Mon Apr 24 09:34:47 CEST 2023**
> Updated 2 taps (homebrew/core and homebrew/cask).
> ==> New Formulae
> ares                                                   millet                                                 poke                                                   vulkan-tools
> ==> Outdated Formulae
> wimlib                                                                                                         yamllint
> 
> You have 2 outdated formulae installed.
> You can upgrade them with brew upgrade
> or list them with brew outdated.
> ==> Upgrading 2 outdated packages:
> wimlib 1.13.6 -> 1.14.0
> yamllint 1.30.0 -> 1.31.0
> ==> Fetching wimlib
> ==> Downloading https://ghcr.io/v2/homebrew/core/wimlib/manifests/1.14.0
> ######################################################################## 100.0%
> ==> Downloading https://ghcr.io/v2/homebrew/core/wimlib/blobs/sha256:95e398be941a4eb36e3cec7d96641d4fdeb299eaa46fd2cdc04bc13a1a7a6471
> ==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/blobs/sha256:95e398be941a4eb36e3cec7d96641d4fdeb299eaa46fd2cdc04bc13a1a7a6471?se=2023-04-24T07%3A45%3A00Z&sig=tJs3Da8rilhUZuX3r2LiMskPWQDPaCK60mCBup
> ######################################################################## 100.0%
> ==> Fetching yamllint
> ==> Downloading https://ghcr.io/v2/homebrew/core/yamllint/manifests/1.31.0
> ######################################################################## 100.0%
> ==> Downloading https://ghcr.io/v2/homebrew/core/yamllint/blobs/sha256:53f3c7903eaeb3d30cbbcd274543630b4bca01b4b96bd3f01d6a9799c7766248
> ==> Downloading from https://pkg-containers.githubusercontent.com/ghcr1/blobs/sha256:53f3c7903eaeb3d30cbbcd274543630b4bca01b4b96bd3f01d6a9799c7766248?se=2023-04-24T07%3A45%3A00Z&sig=YZI2K9rUc4CTh8CEkxtfhswmnWCw3a%2BVHBqF
> ######################################################################## 100.0%
> ==> Upgrading wimlib
>   1.13.6 -> 1.14.0
> 
> ==> Pouring wimlib--1.14.0.arm64_ventura.bottle.tar.gz
> 🍺  /opt/homebrew/Cellar/wimlib/1.14.0: 64 files, 3.0MB
> ==> Running `brew cleanup wimlib`...
> Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
> Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
> Removing: /opt/homebrew/Cellar/wimlib/1.13.6... (64 files, 3.0MB)
> ==> Upgrading yamllint
>   1.30.0 -> 1.31.0
> 
> ==> Pouring yamllint--1.31.0.arm64_ventura.bottle.tar.gz
> 🍺  /opt/homebrew/Cellar/yamllint/1.31.0: 821 files, 10.2MB
> ==> Running `brew cleanup yamllint`...
> Removing: /opt/homebrew/Cellar/yamllint/1.30.0... (850 files, 10.8MB)
> Removing: /Users/jocebal/Library/Caches/Homebrew/yamllint--1.30.0... (2.9MB)
> **Mon Apr 24 09:36:06 CEST 2023**

Speed test FYI

> jocebal@jocebal-mac ~ % speedtest-cli
> Retrieving speedtest.net configuration...
> Testing from Telefonica de Espana Static IP (81.XX.XXX.XXX)...
> Retrieving speedtest.net server list...
> Selecting best server based on ping...
> Hosted by Airsip telecomunicaciones (Valencia) [237.71 km]: 12.204 ms
> Testing download speed................................................................................
> Download: 601.19 Mbit/s
> Testing upload speed......................................................................................................
> Upload: 268.06 Mbit/s

Any idea?

Regards

**Reply by SMillerDev** [reply]
Is there an antivirus tool that could be interfering with unpacking or moving the files?

**Reply by jorgeceballosgarcia** [reply]
Cybereason ActiveProbe

Is it possible that it could interfere?

**Reply by SMillerDev** [reply, marked as answer]
Yeah, that's very likely. Downloading a bunch of binaries like package managers do tends to scare virusscanners.
